HGCalAnalysis/plugins/HGCalAnalysis.cc

```diff
@@ -836,7 +921,75 @@ HGCalAnalysis::analyze(const edm::Event& iEvent, const edm::EventSetup& iSetup)
 		multiclus_slopeY.push_back(multiClusters[i].y());
 		multiclus_cluster2d.push_back(cl2dIndices);
 		multiclus_cl2dSeed.push_back(cl2dSeed);
+		multiclus_firstLay.push_back(*layers.begin());
+		multiclus_lastLay.push_back(*layers.rbegin());
+		multiclus_NLay.push_back(layers.size());
+		if(multiClusters[i].size()<3)
+		  	{
+			multiclus_pcaAxisX.push_back(-2.);
```

Comment:
For a multicluster to be stored there is a requirement on the number of 2D clusters it has, [here](https://github.com/cms-sw/cmssw/blob/master/RecoLocalCalo/HGCalRecAlgos/src/HGCalDepthPreClusterer.cc#L68). The [default value](https://github.com/cms-sw/cmssw/blob/master/RecoLocalCalo/HGCalRecProducers/python/hgcalLayerClusters_cfi.py#L21) is three, so one would only expect multiclusters with at least four 2D clusters. 

I sometimes alter this to keep all the multiclusters so keeping this default seems sensible to me. (As an aside, the minClusters label is a bit misleading since the cut is implemented as a strictly greater than - we should change it at some point).